crates/host/src/module.rs

```diff
@@ -119,6 +129,60 @@ pub trait PlruCache {
     }
 }
 
+#[cfg(not(test))]
+/// one hundred giga ops
+pub const WASM_METERING_LIMIT: u64 = 100_000_000_000;
+
+#[cfg(test)]
+/// ten mega ops.
+/// We don't want tests to run forever, and it can take several minutes for 100 giga ops to run.
+pub const WASM_METERING_LIMIT: u64 = 10_000_000;
+
+/// Generate an engine with a wasm compiler
+/// and Metering (use limits) in place.
+pub fn make_compiler_engine() -> Engine {
+    let cost_function = |_operator: &wasmparser::Operator| -> u64 { 1 };
+    // @todo 100 giga-ops is totally arbitrary cutoff so we probably
+    // want to make the limit configurable somehow.
+    let metering = Arc::new(Metering::new(WASM_METERING_LIMIT, cost_function));
+    // the only place where the wasm compiler engine is set
+    let mut compiler = Cranelift::default();
+    compiler.canonicalize_nans(true).push_middleware(metering);
+    Engine::from(compiler)
+}
+
+/// Generate a runtime `Engine` without compiler suitable for iOS.
+/// Useful for re-building an iOS Module from a preserialized WASM Module.
+pub fn make_ios_runtime_engine() -> Engine {
```

Comment:
Should these be behind a platform flag so they aren't available everywhere? Or are they valid anywhere but just recommended for iOS? 

Actually I think even in the second case, if they have _ios_ in the name I'd be in favour of them being behind a platform condition

Reply:
Understood, I still think they probably ought to be but as this was just moved code it can wait for a less busy time :)

Reply:
Another limiting factor is that where they're used in `holochain` is in an `if..else` assignment, so I think we can't hide a part of it behind a feature.